public/app/features/dashboard-scene/serialization/transformSceneToSaveModel.ts

```diff
@@ -186,21 +176,36 @@ export function gridItemToPanel(gridItem: DashboardGridItem, isSnapshot = false)
     return libraryVizPanelToPanel(gridItem.state.body, { x, y, w, h });
   }
 
-  if (!(gridItem.state.body instanceof VizPanel)) {
+  let gridItem_ = gridItem;
+
+  // If we're saving while the panel editor is open, we need to persist those changes in the panel model
+  if (
+    sceneState &&
```

Comment:
Can't we get this state from the gridItem using `getDashboardSceneFor`? With that, we would avoid to pass down the dashboard scene state. WDYT?

Reply:
Yeah, I agree that it's ugly passing the state down through the functions, but I wanted to avoid calling `getDashboardSceneFor` multiple times

Reply:
Looking at the function 
https://github.com/grafana/grafana/blob/7d84ae04aac74d77aef0f05d5036c5f1884dec82/public/app/features/dashboard-scene/utils/utils.ts#L185

I don't see any issue with calling it multiple times, it is just getting the root. Am I missing something here?

FYI, we're discussing it with the team, and @Sergej-Vlasov is trying to refactor it to use `getDashboardSceneFor `. We can always move it back to this approach if there is something we're missing.